### DISCUSSION TITLE
OT: Kindle catalog?

### DISCUSSION THREAD

**Discussion by giovannicoppola** [mkb79/audible-cli#140] [General]
Hi @mkb79 I hope you are well!! I use the audible alfred app every day and others have dowloaded it too! 
I am thinking to create a similar one for kindle books (i.e. to search my library, and also search the Kindle main catalog). 

Are you aware of a way to interrogate the kindle catalog as you do for audible? I found a static catalog [here](https://www.kaggle.com/snathjr/kindle-books-dataset), but of course it would be great to get updated information directly from Amazon. At a minimum, I would like to search title/author and get the ASIN back. Thanks!!

**Comment by mkb79** [comment]
Hi. I've created a [kindle package](https://github.com/mkb79/kindle) some years ago. These let you fetch your own library and can download your own books.
The Kindle app support browsing the catalog since some time. I'm think you can use my package for authorize you requests to the kindle API. But you have to find out the catalog endpoint before. 

If you use Kindle for iPhone or iPad I can tell you a tool which let you record the requests the kindle app does. So you can replay these.

**Reply by giovannicoppola** [reply]
thanks! this sounds great. I just tried it and I keep getting CAPTCHAs. Have you tried it recently?

**Reply by giovannicoppola** [reply]
I am thinking I should be able to get catalog details from Goodreads too?

**Reply by mkb79** [reply]
I does not recognize that you had post an answer. I does not have tried the code recently. But the login/device registration is the same as for the Audible service. Maybe these part must be reworked to match the Audible code base. 

I does not know if the Goodreads API has an endpoint for this. But the API docs are online. You can Google for them.
